### ISSUE TITLE
`Config.extra` is messed-up when using multiple inheritance/mixins

### ISSUE THREAD

**Issue by YaraslauZhylko** [samuelcolvin/pydantic#392]
# Bug

* OS: Ubuntu 18.04
* Python version: 3.7.1
* Pydantic version: 0.19

`extra` Config option does not seem to propagate correctly when using multiple inheritance/mix-ins.
Other options seem to work OK.

```py
import pydantic


class Base(pydantic.BaseModel):

    field: str

    class Config:
        allow_mutation = False
        use_enum_values = True
        extra = pydantic.Extra.forbid


class Mixin(pydantic.BaseModel):

    another_field: int = 0


class SubClass(Mixin, Base):

    yet_another_field: float = 0.1


assert pydantic.BaseModel.__config__.allow_mutation is True
assert pydantic.BaseModel.__config__.use_enum_values is False
assert pydantic.BaseModel.__config__.extra == pydantic.Extra.ignore

assert Base.__config__.allow_mutation is False
assert Base.__config__.use_enum_values is True
assert Base.__config__.extra == pydantic.Extra.forbid

assert SubClass.__config__.allow_mutation is False
assert SubClass.__config__.use_enum_values is True
assert SubClass.__config__.extra == pydantic.Extra.forbid  # FAILS HERE; RESET TO `ignore`
```

Potentially caused by the way legacy `allow_extra` / `ignore_extra` are supported.

**Comment by YaraslauZhylko** [comment]
Replacing `extra` with `allow_extra = False` and `ignore_extra = False` seems to work as expected:

```py
class Config:
        allow_mutation = False
        use_enum_values = True
        allow_extra = False
        ignore_extra = False
        # extra = pydantic.Extra.forbid
```

**Comment by YaraslauZhylko** [comment]
Some fancy "real runtime code":

```py
print("====== BASE CLASS =====")

print("* Normal field set:")
Base(field="test")
print("OK")

print("* Mutantion:")
try:
    Base(field="test").field = "different value"
except TypeError as err:
    print("Expected error:", err)
else:
    print(">>>>>>>>>> NO ERROR! <<<<<<<<<<")

print("* Extra:")
try:
    Base(field="test", extra_field="extra")
except pydantic.ValidationError as err:
    print("Expected error:", err)
else:
    print(">>>>>>>>>> NO ERROR! <<<<<<<<<<")

print()
print("====== SUBCLASS =====")

print("* Normal field set:")
SubClass(field="test", another_field=1)
print("OK")

print("* Mutantion:")
try:
    SubClass(field="test", another_field=1).field = "different value"
except TypeError as err:
    print("Expected error:", err)
else:
    print(">>>>>>>>>> NO ERROR! <<<<<<<<<<")

print("* Extra:")
try:
    SubClass(field="test", another_field=1, extra_field="extra")
except pydantic.ValidationError as err:
    print("Expected error:", err)
else:
    print(">>>>>>>>>> NO ERROR! <<<<<<<<<<")
```

**Comment by samuelcolvin** [comment]
Thanks for reporting. 

Would you consider submitting a PR to fix? I guess we should skip `allow_extra` `ignore_extra` from config inheritance if `extra` is set.

**Comment by YaraslauZhylko** [comment]
Will try.
Probably not today though...

**Comment by samuelcolvin** [comment]
Thanks.

Not today is fine.

**Comment by YaraslauZhylko** [comment]
@samuelcolvin PR's ready: https://github.com/samuelcolvin/pydantic/pull/394

**Comment by samuelcolvin** [comment]
fixed by #394
